app/controllers/admin/merchants_controller.rb

```diff
@@ -14,10 +16,17 @@ def edit
   end
   
   def update
+    @merchant = Merchant.find(params[:id])
     if @merchant.update(merchant_params)
-      redirect_to admin_merchant_path, flash: { success: "Merchant information updated successfully." }
+      if !params[:merchant][:status].nil?
+        redirect_to admin_merchants_path
+      else 
+        flash[:success] = 'Merchant information updated successfully.'
+        redirect_to admin_merchant_path(@merchant)
+      end
     else
-      render :edit_name
+      flash[:error] = "Merchant must have a name."
```

Comment:
I was gonna put this in the applications controller 

 private
    def error_message(errors)
      errors.full_messages.join(', ')
    end
    
    it may work here

Reply:
Would that functionality inherit and we could call it in some way? Flash messages are still a bit of a weak spot for me tbh. I'd appreciate a lil walkthrough if you're down!

Reply:
It might, let me do a little research and then I would be down to walk you through it in a bit! I have an errand to run this morning